File: src/main/java/com/company/githubintegration/entity/Repository.java
package com.company.githubintegration.entity;

import io.jmix.core.entity.annotation.JmixGeneratedValue;
import io.jmix.core.entity.annotation.JmixId;
import io.jmix.core.metamodel.annotation.InstanceName;
import io.jmix.core.metamodel.annotation.JmixEntity;

@JmixEntity
public class Repository {

    @JmixGeneratedValue
    @JmixId
    private Long id;

    @InstanceName
    private String name;

    private Long stargazersCount;

    private UserInfo owner;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Long getStargazersCount() {
        return stargazersCount;
    }

    public void setStargazersCount(Long stargazersCount) {
        this.stargazersCount = stargazersCount;
    }

    public UserInfo getOwner() {
        return owner;
    }

    public void setOwner(UserInfo owner) {
        this.owner = owner;
    }
}

File: frontend/src/jmix/entities/Repository.ts
import { UserInfo } from "./UserInfo";
import {HasId} from "@haulmont/jmix-react-core";
export class Repository {
  static NAME = "Repository";
  id?: string;
  name?: string | null;
  stargazersCount?: any | null;
  owner?: UserInfo | null;
}
export type RepositoryViewName = "_base" | "_instance_name" | "_local";
export type RepositoryView<V extends RepositoryViewName> = V extends "_base"
  ? Pick<Repository, "id" | "name">
  : V extends "_instance_name"
  ? Pick<Repository, "id" | "name">
  : never;